File: day-4/2.ex
File.read!("./input.txt")
|> String.split("\n")
|> Enum.map(fn line ->
  line
  |> String.split(",")
  |> Enum.map(fn l ->
    [left, right] =
      l
      |> String.split("-")
      |> Enum.map(&String.to_integer/1)

    Range.new(left, right)
  end)
  |> List.to_tuple()
end)
|> Enum.reject(fn {left, right} ->
  Range.disjoint?(left, right)
end)
|> length()
|> IO.inspect()

File: day-04/2.ts
export {};

const input = await Deno.readTextFile("./input.txt").then((input) =>
  input.split("\n")
);

let cont = 0;

input.map((line) => {
  const [left, right] = line.split(",");
  const [left1, left2] = left.split("-").map((n) => parseInt(n));
  const [right1, right2] = right.split("-").map((n) => parseInt(n));

  if (left1 >= right1 && left1 <= right2) {
    cont++;
  } else if (left1 <= right1 && right1 <= left2) {
    cont++;
  } else if (left2 >= right1 && left2 <= right2) {
    cont++;
  } else if (right2 >= left1 && right2 <= left2) {
    cont++;
  }
});

console.log(cont);
